crates/host/src/wasmbus/mod.rs

```diff
@@ -2911,10 +2911,8 @@ impl Host {
         );
 
         let Some(mut component_spec) = self.get_component_spec(source_id).await? else {
-            // If the component spec doesn't exist, the link is deleted
-            return Ok(CtlResponse::<()>::success(
-                "successfully deleted link (spec doesn't exist)".into(),
-            ));
+            // If the component spec doesn't exist, then the link does not exist ;-)
+            return Ok(CtlResponse::error("link does not exist"));
```

Comment:
We discussed this for a while in Slack (apologies for having context there that didn't make it in the GitHub issue) https://wasmcloud.slack.com/archives/CS38R7N9Y/p1732544510210499?thread_ts=1732459582.627589&cid=CS38R7N9Y

I think this should actually still be `CtlResponse::success` only because I'm interested in keeping the idempotency here where deleting a link that existed or didn't exist still returns a successful return, because the link is now deleted